### PR TITLE
Transparent protection for clothing [READY]

### LINKED ISSUE #41618: fire burning clothing behind fireproof clothes 2: electirc boogaloo
>be op >elite hardsuit in combat mode >start fire >Your syndicate mask starts to fall apart! >die

### LINKED ISSUE #41831: Fire burns mask through cap suit
The FULLY FIREPROOF SUIT doesn't mean shit if your mask can burn off, I'm at least fairly sure none of the standard masks on the station are fireproof either. I can't breathe fire, Nanotrasen!!!

### LINKED ISSUE #42005: Fire burns mask through cap suit
Reference #41831 and #41618 for more details, but this is still a problem with the captain's hardsuit.

### PR DESCRIPTION
**Pull request by 4dplanner**
:cl: 
fix: Some items cover equipment while still letting you see it
fix: in particular, this stops clothes burning off underneath drake armor
/:cl:

Allows you to fashionably wear drake armor and be on fire without all your clothes burning off.

This can't just be done by flags_inv as drake armour has visible holes, but still protects from fire totally. More generally, I believe this makes sense for helmets with glass transparent enough to look into and so forth, so I think it's worth adding a seperate set of flags for. 

This could apply to captain's, but I don't think it really makes sense. Adding it would just be fixing a gameplay issue with a bug, which would be lame.

closes #42094

### CODE REVIEW

**Comment by 4dplanner** [comment]
@ShizCalev I know you were planning on adding respirators, but it's pretty annoying in the meantime. In addition, Drake armor really needs a system like this, so I thought I might as well chuck in some mask protection for the captain in the meantime.

**Comment by 4dplanner** [comment]
It also doesn't really make sense for your clothes to be burning off while you're totally unharmed

**Comment by 4dplanner** [comment]
I was joking, I know what you meant

**Comment by 4dplanner** [comment]
I've actually decided this doesn't make sense for captain's hardsuit, so this is just for drake armor/firesuit if that turns out to be broken.

**Comment by 4dplanner** [comment]
Actually this'd only be used for like one thing, not worth putting a flag on everything

**Comment by Bod9001** [comment]
what's the point of the captain's hard suit being fireproof If your mask burns off?

**Comment by 4dplanner** [comment]
> what's the point of the captain's hard suit being fireproof If your mask burns off?

it's currently shit, I agree, but there are plans to fix it by adding fireproof masks.

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -48,6 +48,7 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 
 	//Since any item can now be a piece of clothing, this has to be put here so all items share it.
 	var/flags_inv //This flag is used to determine when items in someone's inventory cover others. IE helmets making it so you can't see glasses, etc.
+	var/transparent_protection = 0 //you can see someone's mask through their transparent visor, but you can't reach it
```
Comment (Cyberboss) [inline code/game/objects/items.dm]:
```suggestion
 	var/transparent_protection = NONE //you can see someone's mask through their transparent visor, but you can't reach it
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**